pkg/dynamiclistener/server.go

```diff
@@ -596,3 +646,35 @@ func (ln tcpKeepAliveListener) Accept() (c net.Conn, err error) {
 	tc.SetKeepAlivePeriod(3 * time.Minute)
 	return tc, nil
 }
+
+func lookupTLSVersion(stringMinTLSVersion string) (uint16, error) {
+	tlsVersionKeys := getKeysFromMap(tlsVersions)
+	if val, ok := tlsVersions[stringMinTLSVersion]; ok {
+		return val, nil
+	}
+	return 0, fmt.Errorf("Invalid minimal TLS version [%s], must be one of: %s", stringMinTLSVersion, strings.Join(tlsVersionKeys, " "))
+}
+
+func lookupTLSCiphers(stringTLSCiphers string) ([]uint16, error) {
+	tlsCipherSuitesKeys := getKeysFromMap(tlsCipherSuites)
+	sliceTLSCiphers := strings.Split(stringTLSCiphers, ",")
+
+	var TLSCiphers []uint16
+	for _, TLSCipher := range sliceTLSCiphers {
+		val, ok := tlsCipherSuites[TLSCipher]
+		if !ok {
+			return []uint16{}, fmt.Errorf("Unsupported cipher [%s], must be one or more from: %s", TLSCipher, strings.Join(tlsCipherSuitesKeys, " "))
+		}
+		TLSCiphers = append(TLSCiphers, val)
+	}
+	return TLSCiphers, nil
+}
+
+func getKeysFromMap(input interface{}) []string {
```

Comment:
I just had a question as to why we have input as `interface{}`? Since both maps are of same type, we can just use it and do, 
```
keys = []string 
for key := range map {
     keys = append(keys, key) 
}
return keys 
```

Reply:
Was made as generic function for re-use but can be done when necessary.